apps/api/src/domain/models/resource-deleted.error.ts

```diff
@@ -0,0 +1,13 @@
+import { InternalError } from '../../lib/errors/InternalError';
+import formatResourceCompositeIdentifier from '../../queries/presentation/formatAggregateCompositeIdentifier';
+import { ResourceCompositeIdentifier } from '../types/ResourceCompositeIdentifier';
+
+export default class ResourceDeletedError extends InternalError {
```

Comment:
Oh this is really interesting. We might want 'ResourceNotFound' error instead. Once the resource is deleted, it should appear to the user that it does not exist. That said, internally we will want to test the case where the resource used to exist but is now deleted (archived).

Reply:
done